src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -617,6 +617,13 @@ private function addValidationSection(ArrayNodeDefinition $rootNode)
                                 ->then(function ($v) { return (string) $v; })
                             ->end()
                         ->end()
+                        ->arrayNode('mappings')
+                            ->prototype('array')
+                                ->children()
+                                    ->scalarNode('file')->end()
```

Comment:
``mapping`` would become a singular in such case though, as you would have a single node for it though.

I agree with @xabbuh though. The prototyped node should not be ``mappings`` if you only configure either ``file`` or ``dir`` in it